foreman.te

```diff
@@ -171,190 +188,156 @@ require{
-        # puppet_port_t before passenger is initialized.
-        read_files_pattern(httpd_t, puppet_etc_t, puppet_etc_t)
-        read_lnk_files_pattern(httpd_t, puppet_etc_t, puppet_etc_t)
-        # http://projects.theforeman.org/issues/9805
-        selinux_validate_context(passenger_t)
+    tunable_policy(`foreman_rails_can_connect_puppetmaster', `
+        corenet_tcp_connect_puppet_port(foreman_rails_t)
     ')
 ')
 
+# Sending of email reports via sendmail
 optional_policy(`
-    tunable_policy(`passenger_can_connect_puppetmaster', `
-        corenet_tcp_connect_puppet_port(httpd_t)
-        corenet_tcp_connect_puppet_port(passenger_t)
-    ')
+    mta_send_mail(foreman_rails_t)
 ')
 
-optional_policy(`
-    tunable_policy(`passenger_can_connect_http', `
-        corenet_tcp_connect_http_port(httpd_t)
-        corenet_tcp_connect_http_port(passenger_t)
-    ')
+# Sending of email reports via SMTP directly
+tunable_policy(`foreman_rails_can_connect_smtp',`
+    corenet_tcp_connect_smtp_port(foreman_rails_t)
 ')
 
-# Puppetmaster touches file contexts of some files which causes issues when apache2 httpd
-# is started from command line (under unconfined_u). This rule makes sure it does not
-# go into unconfined_u. This is only needed for RHEL6 (upstart). This rule MUST NOT be
-# in an optional_policy block!
-domain_obj_id_change_exemption(passenger_t)
-
+# Connecting to LDAP
 optional_policy(`
-    # Allow sending of email reports.
-    mta_send_mail(passenger_t)
-')
-
-tunable_policy(`passenger_can_connect_smtp',`
-    corenet_tcp_connect_smtp_port(passenger_t)
+    tunable_policy(`foreman_rails_can_connect_ldap', `
+        corenet_tcp_connect_ldap_port(foreman_rails_t)
+    ')
 ')
 
-optional_policy(`
-    mysql_stream_connect(passenger_t)
-    mysql_list_db(passenger_t)
+# Allow Foreman to write to the SQlite databases
+tunable_policy(`foreman_rails_can_manage_sqlite',`
+    read_files_pattern(foreman_rails_t, foreman_db_t, foreman_db_t)
+    write_files_pattern(foreman_rails_t, foreman_db_t, foreman_db_t)
 ')
 
-optional_policy(`
-    hostname_exec(passenger_t)
+# General "connect everywhere" rule (disabled by default)
+tunable_policy(`foreman_rails_can_connect_all',`
+    corenet_tcp_connect_all_ports(foreman_rails_t)
 ')
 
-optional_policy(`
-    # Transition to puppet_master
-    corecmd_search_bin(passenger_t)
-    domtrans_pattern(passenger_t, puppetmaster_exec_t, puppetmaster_t)
-')
+#################
+# TODO
+#
+# system_u:system_r:unconfined_service_t:s0 foreman 1227 0.0  0.0 4356 584 ?     Ss   10:26   0:00 /usr/bin/scl enable tfm sidekiq -e production -r /usr/share/foreman/extras/dynflow-sidekiq.rb 
+# system_u:system_r:unconfined_service_t:s0 foreman 1230 0.0  0.0 113284 1472 ?  S    10:26   0:00  \_ /bin/bash /var/tmp/sclPZHShi
+# system_u:system_r:unconfined_service_t:s0 foreman 1671 0.3  4.2 972624 545000 ? Sl  10:26   1:18      \_ sidekiq 5.2.7  [0 of 5 busy]
```

Comment:
Now that we have the wrapper, this is no longer an issue, right?

Reply:
LOL I left these notes here, removed :-)